core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -117,11 +117,28 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.DELETE_RECORDS => handleDeleteRecordsRequest(request)
         case ApiKeys.INIT_PRODUCER_ID => handleInitProducerIdRequest(request)
         case ApiKeys.OFFSET_FOR_LEADER_EPOCH => handleOffsetForLeaderEpochRequest(request)
-        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleAddPartitionToTxnRequest(request)
-        case ApiKeys.ADD_OFFSETS_TO_TXN => handleAddOffsetsToTxnRequest(request)
-        case ApiKeys.END_TXN => handleEndTxnRequest(request)
-        case ApiKeys.WRITE_TXN_MARKERS => handleWriteTxnMarkersRequest(request)
-        case ApiKeys.TXN_OFFSET_COMMIT => handleTxnOffsetCommitRequest(request)
+        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleWithInterBrokerVersionAtLeast(KAFKA_0_11_0_IV0, request, (error:Errors) => {
```

Comment:
@dguy I think there are a couple ways to do it. If you look in `ApiVersionsResponse`, we have a static response  `API_VERSIONS_RESPONSE` which contains all of the APIs that the broker supports (without regard to inter broker protocol). One option might be to turn that into a map which is keyed by the inter broker protocol. Maybe the minimum inter-broker protocol could then be added as a field to `ApiKeys`?